### PR TITLE
Fix @mention recognition in Townsquare

### LINKED ISSUE #6298: Townsquare feed should recognize links
### What 
Townsquare feed should recognize links - make link clickable

### Why
not clickable

Screen Shot 2020-03-25 at 8 51 18 AM

### PR DESCRIPTION
**Pull request by sebastiantf**
##### Description

The existing regex to identify @mentions was also identifying the same that could occur in URLs like from medium: https://medium.com/@shevski/how-decentralised-are-you-a6539eeb27ff

The regex has been modified to correctly identify intended @mentions and not those in URLs.

##### Refers/Fixes

Fixes #6298 

##### Testing

Original input to status update:
```
@shevski @shevski @shevski

Although the article is published over one year ago. It’s a must read for anyone involved in tech related businesses. https://medium.com/@shevski/how-decentralised-are-you-a6539eeb27ff
```

**Before:**

Screenshot 2020-04-22 at 1 15 39 AM

Also see: https://github.com/gitcoinco/web/issues/6298#issuecomment-612713914

**After:**

Screenshot 2020-04-22 at 1 16 07 AM

### CODE REVIEW

File: app/assets/v2/js/activity.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ $(document).ready(function() {
 
   var linkify = function(new_text) {
     new_text = new_text.replace(/(?:^|\s)#([a-zA-Z\d-]+)/g, ' <a href="/?tab=search-$1">#$1</a>');
-    new_text = new_text.replace(/\B@([a-zA-Z0-9_-]*)/g, ' <a data-usercard="$1" href="/profile/$1">@$1</a>');
+    new_text = new_text.replace(/(^|\s)@([a-zA-Z0-9_-]*)/g, '$1<a data-usercard="$2" href="/profile/$2">@$2</a>');
```
Comment (octavioamu) [inline app/assets/v2/js/activity.js:11]: is this $1 right?

Comment (octavioamu) [reply app/assets/v2/js/activity.js:11]: I believe that $1 was fat finger https://regex101.com/r/5H3V5y/1 but tell me if Im wrong

Comment (sebastiantf) [reply app/assets/v2/js/activity.js:11]: Yes it is. It copies over the first capturing group which checks for beginning or whitespace (`(^|\s)`)

Comment (sebastiantf) [reply app/assets/v2/js/activity.js:11]: This is what you'd get without $1 or any space.

Screenshot 2020-04-22 at 1 56 16 AM

Comment (octavioamu) [reply app/assets/v2/js/activity.js:11]: ohh totally make sense, thanks for explaining

**Pull request merged**